File: algorithms/cpp/minimumMovestoEqualArrayElementsII/minimumMovestoEqualArrayElementsII.cpp
// Source: https://leetcode.com/problems/minimum-moves-to-equal-array-elements-ii/
// Author: Miao Zhang
// Date:   2021-02-14

class Solution {
public:
    int minMoves2(vector<int>& nums) {
        int n = nums.size();
        nth_element(nums.begin(), nums.begin() + n / 2, nums.end());
        int res = 0;
        for (auto num: nums) {
            res += abs(num - nums[n / 2]);
        }
        return res;
    }
};

File: algorithms/python/minimumMovestoEqualArrayElementsII/minimumMovestoEqualArrayElementsII.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/minimum-moves-to-equal-array-elements-ii/
# Author: Miao Zhang
# Date:   2021-02-14

class Solution:
    def minMoves2(self, nums: List[int]) -> int:
        median = sorted(nums)[len(nums) // 2]
        return sum([abs(n - median) for n in nums])
